framework/src/constraints/MortarConstraint.C

```diff
@@ -15,243 +15,442 @@
 #include "MooseVariable.h"
 #include "NearestNodeLocator.h"
 #include "PenetrationLocator.h"
+#include "AutomaticMortarGeneration.h"
 
 #include "libmesh/quadrature.h"
 
-template <>
-InputParameters
-validParams<MortarConstraint>()
-{
-  InputParameters params = validParams<Constraint>();
-  params.addRequiredParam<std::string>("interface", "The name of the interface.");
-  params.addRequiredParam<VariableName>("master_variable", "Variable on master surface");
-  params.addParam<VariableName>("slave_variable", "Variable on master surface");
-  return params;
-}
+defineADBaseValidParams(
+    MortarConstraint,
+    MortarConstraintBase,
+    params.addParam<BoundaryID>("master_boundary_id", "The id of the master boundary sideset.");
+    params.addParam<BoundaryID>("slave_boundary_id", "The id of the slave boundary sideset.");
+    params.addParam<SubdomainID>("master_subdomain_id", "The id of the master subdomain.");
+    params.addParam<SubdomainID>("slave_subdomain_id", "The id of the slave subdomain.");
```

Comment:
In general, we always read in "Name"s instead of "ID"s. The reason is that "Name"s handle ID cases seamlessly.